mustang_codegen/lib/src/state_generator/screen_state_generator.dart

```diff
@@ -93,14 +93,17 @@ class ScreenStateGenerator extends Generator {
               'modelStr': 'disposed',
             });
           }
+          MustangStore.delete<$stateName>();
           MustangRouteObserver.getInstance().unsubscribe(this);
           super.dispose();
         }
         
         /// Called when the screen associated with this state has been popped off.
         @override
         void didPop() {
-          // TODO: implement didPop
+          Timer(const Duration(seconds: 1), () {
```

Comment:
scheduling microtask for disposing will not help as there is no guarantee that cleanup tasks related to popping route are queued before the dispose call. Since timer events are processed after microtasks, we can be sure while disposing the state instance it is not in use.